.github/workflows/tests.yaml

```diff
@@ -11,18 +11,51 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+
       - uses: actions/setup-node@v2
         with:
           node-version: '16'
+
       - uses: actions/cache@v2
         with:
           path: '**/node_modules'
           key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
       - name: install dependencies
         run: yarn install --immutable --network-concurrency=1
+
       - name: run lint
         run: yarn run --silent lint -- --max-warnings 0
+
       - name: set host bucketclient.testing.local
         run: sudo bash -c 'echo "127.0.0.1 bucketclient.testing.local" >> /etc/hosts'
+
       - name: run test
         run: yarn run --silent test
+
+  go-client-test:
+    name: Test the Go client
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash
+        working-directory: go-client/
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.23'
+
+      - name: Install ginkgo and golangci-lint
+        run: |
+          go install github.com/onsi/ginkgo/v2/ginkgo@v2.20.2
```

Comment:
not really needed, can be run with `go test` to avoid installing extra package in CI

Reply:
Indeed, `go test` works too, I could use it. But I realized that even with `go test -v` it doesn't show the details of each test run, which I find nice to be able to see which tests are run. It basically behaves like `ginkgo run` (without `-v`). I don't know if that's a sufficiently good reason not to use `go test`.